Project-18-Food-order-app/src/components/Checkout/index.jsx

```diff
@@ -0,0 +1,162 @@
+    handleInputChange: handleNameChange,
+    handleInputBlur: handleNameBlur,
+    hasError: hasNameError,
+  } = useInput("", (value) => isAllLetter(value));
+
+  const cartTotal = cartData.reduce(
+    (totalPrice, item) => totalPrice + +item.price * item.quantity,
+    0
+  );
+
+  const handleCloseButtonClick = () => {
+    hideCheckout();
+    onClose();
+  };
+
+  const {
+    data,
+    isLoading: isSending,
+    error,
+    sendRequest,
+  } = useHttp(`${URL}orders`, requestConfig);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (hasEmailError || hasNameError) return;
+
+    const fd = new FormData(e.target);
+
+    const customerData = Object.fromEntries(fd.entries());
+
+    sendRequest(
+      JSON.stringify({
+        order: {
+          items: cartData,
+          customer: customerData,
+        },
+      })
+    );
+  };
+
+  let actions = (
+    <>
+      <Button type="button" textOnly onClick={handleCloseButtonClick}>
+        Close
+      </Button>
+      <Button type="submit">Submit Order</Button>
+    </>
+  );
+
+  if (isSending) {
+    actions = <span>Sending order data...</span>;
+  }
+
+  useEffect(() => {
+    if (data && !error) {
+      hideCheckout();
+      showSuccessMessageDialog();
+      clearCartData();
+    }
+  }, [data, error, clearCartData, hideCheckout, showSuccessMessageDialog]);
+
+  return (
+    <>
+      <div className={style.cart}>
+        <h2 className="modal-title">Checkout</h2>
+        <p>Total Amount: {currencyFormatter.format(cartTotal)}</p>
+        <div className={style["user-checkout-form"]}>
+          <form onSubmit={handleSubmit}>
+            <Input
+              id="name"
+              label="Full Name"
+              type="text"
+              value={nameValue}
+              onChange={handleNameChange}
+              onBlur={handleNameBlur}
+              error={hasNameError ? "Invalid name" : null}
+              required
+            />
+
+            <Input
+              id="email"
+              label="E-Mail Address"
+              type="email"
+              value={emailValue}
+              onChange={handleEmailChange}
+              onBlur={handleEmailBlur}
+              error={hasEmailError ? "Please enter a valid email." : null}
+              required
+            />
+
+            <Input label="Street" id="street" type="text" required />
+
+            <div className={style.address}>
+              <Input
+                label="Postal Code"
+                type="text"
+                id="postal-code"
+                required
+              />
+              <Input label="City" id="city" type="text" required />
+            </div>
+
+            {error ? (
+              <Error title="Failed to submit order" message={error} />
+            ) : null}
+
+            <div className="modal-actions">{actions}</div>
+          </form>
+        </div>
+      </div>
+    </>
+  );
+}
+
+Checkout.propTypes = {
+  onClose: PropTypes.func,
```

Comment:
@sknepali You have used PropTypes to type-check your props, which is good. However, you can make it more specific. For instance, instead of PropTypes.func for onClose, you can use PropTypes.func.isRequired to ensure that this function is always passed.